web_infrastructure/auth0_user.py

```diff
@@ -0,0 +1,168 @@
+    user_id: userid
+
+# Create new auth0 user for a given domain
+- auth0_user:
+    state: present
+    auth0_token: token
+    auth0_domain: domain
+    user_data:
+      connection: Username-Password-Authentication
+      email: john.doe@domain.com
+      password: SecretPassword
+      email_verified: true
+      app_metadata:
+
+      user_metadata:
+        firstName: John,
+        language: en,
+        lastName: Doe,
+
+# Remove all users from given auth0 domain
+- auth0_user:
+    state: delete_all
+    auth0_token: token
+    auth0_domain: domain
+'''
+
+RETURN = '''
+#only defaults
+'''
+
+import sys
+import requests
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            auth0_token=dict(required=True, type='str', aliases=['token']),
+            auth0_domain=dict(required=True, type='str', aliases=['domain']),
+            state=dict(choices=['present', 'absent', 'delete_all']),
+            user_id=dict(required=False, type='str'),
+            user_data=dict(type='dict', required=False, aliases=['body']),
+
+        ),
+        supports_check_mode=False
+    )
+
+    auth0_domain=module.params['auth0_domain']
+    auth0_token=module.params['auth0_token']
+    state=module.params['state']
+    user_id=module.params['user_id']
+    user_data=module.params['user_data']
+
+    if state == 'present' or state == 'absent':
+      if not user_data:
+        module.fail_json(msg='user_data is required parameter for this state')
+
+    if state == 'absent':
+      if not user_id:
+        module.fail_json(msg='user_id is required parameter for this state')
+
+    http_headers = {
+              'content-type': 'application/json',
+              'Authorization': 'Bearer {token}'.format(token=auth0_token)
+    }
+
+    url_params = user_data
+
+    if state == 'present':
+      if user_id:
+        url = 'https://{domain}/api/v2/users/{userid}'.format(domain=auth0_domain, userid=user_id)
+        r = requests.patch(url, json=user_data, headers=http_headers)
+      else:
+        url = 'https://{domain}/api/v2/users'.format(domain=auth0_domain)
+        r = requests.post(url, json=user_data, headers=http_headers)
+    elif state == 'absent':
+      url = 'https://{domain}/api/v2/users/{userid}'.format(domain=auth0_domain, userid=user_id)
+      r = requests.delete(url, headers=http_headers)
+    elif state == 'delete_all':
+      url = 'https://{domain}/api/v2/users'.format(domain=auth0_domain)
+      r = requests.delete(url, headers=http_headers)
+
+    if r.status_code not in [ 200, 201, 202, 203, 204 ]:
+      module.fail_json(msg='Request to Auth0 failed with return code %s, reason: %s'%(r.status_code, r.reason))
+
+    if r.text:
+      result = json.loads(r.text)
+    else:
+      result = "Rest call has not returned any data."
+
+    module.exit_json(results=result)
+
+# import module snippets
+from ansible.module_utils.basic import *
+from ansible.module_utils.urls import *
+
```

Comment:
Please add the following and remove the `import requests`
`from ansible.module_utils.urls import *`

For more information please search for `ansible.module_utils.urls` on http://docs.ansible.com/ansible/developing_modules.html#module-checklist